docs/flows/flows.puml

```diff
@@ -272,20 +271,136 @@ bitdaoadmin <-- insurancecontract: execute proposal
+
+bitdaoadmin -> bitdaogovernance: get roles voting result
+bitdaoadmin <-- bitdaogovernance: pass/fail vote result
+
+bitdaoadmin -> bitdaogovernance: start execution
+executor <-- bitdaogovernance: execute proposal
+executor -> proxyadmin: call governance proxy upgrade to new bitdaogovernance implementation
+
+newpage
+
+autonumber "UpgradeTreasury:[00]"
+
+== Upgrade Treasury ==
+
+bitdaoadmin -> bitdaotreasury: deploy initial treasury contract logic, set governance
+bitdaoadmin -> proxyadmin: deploy admin for treasury proxy
+bitdaoadmin -> proxy: deploy treasury proxy and set logic and admin
+bitdaoadmin -> proxyadmin: set governance executor as admin
+
+
 alice -> bitdaotreasury: deploy new bitdaotreasury implementation
-alice -> executor: deploy new executor implementation
 
-alice -> forum: create forum upgrade proposal
+alice -> forum: create forum upgrade proposal with new bitdaogovernance
+alice <- community: positive/negative feedback
+
+community -> community: delegate role (treasury/developer/all) based votes
+
+alice -> bitdaogovernance: submit upgrade proposal
+community -> bitdaogovernance: roles voting
+
+bitdaoadmin -> bitdaogovernance: get roles voting result
+bitdaoadmin <-- bitdaogovernance: pass/fail vote result
+
+bitdaoadmin -> bitdaogovernance: start execution
+executor <-- bitdaogovernance: execute proposal
+executor -> proxyadmin: call treasury proxy upgrade to new bitdaotreasury implementation
+
+newpage
+
+autonumber "UpgradeVault:[00]"
+
+== Upgrade Vault ==
+
+bitdaoadmin -> vault: deploy initial vault contract logic, set treasury
+bitdaoadmin -> proxyadmin: deploy admin for vault proxy
+bitdaoadmin -> proxy: deploy vault proxy and set logic and admin
+bitdaoadmin -> proxyadmin: set governance executor as admin
+
+
+alice -> vault: deploy new vault implementation
+
+alice -> forum: create forum upgrade proposal with new bitdaogovernance
+alice <- community: positive/negative feedback
+
+community -> community: delegate role (treasury/all) based votes
+
+alice -> bitdaogovernance: submit upgrade proposal
+community -> bitdaogovernance: roles voting
+
+bitdaoadmin -> bitdaogovernance: get roles voting result
+bitdaoadmin <-- bitdaogovernance: pass/fail vote result
+
+bitdaoadmin -> bitdaogovernance: start execution
+executor <-- bitdaogovernance: execute proposal
+executor -> proxyadmin: call vault proxy upgrade to new vault implementation
+
+newpage
+
+autonumber "UpgradeBond:[00]"
+
+== Upgrade Bond ==
+
+bitdaoadmin -> bondcontract: deploy initial bond factory contract logic, set treasury
+bitdaoadmin -> proxyadmin: deploy admin for bond factory proxy
+bitdaoadmin -> proxy: deploy bond factory proxy and set logic and admin
+bitdaoadmin -> proxyadmin: set governance executor as admin
+
+
+alice -> bondcontract: deploy new bond factory contract implementation
+
+alice -> forum: create forum upgrade proposal with new bitdaogovernance
+alice <- community: positive/negative feedback
```

Comment:
nit: double minus sign for a dotted return message line

UML message return lines are usually dotted rather than solid.